trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/TrinoS3ProxyServerPlugin.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server;
+
+import com.google.inject.Module;
+
+public interface TrinoS3ProxyServerPlugin
```

Comment:
Currently the plugin mechanism will provide bindings based solely on Guice. Should we define what methods this would support similarly to the Trino SPI? https://github.com/trinodb/trino/blob/0f210c0decf6a3c52320e677ac95febac0c94fc2/core/trino-spi/src/main/java/io/trino/spi/Plugin.java

Perhaps a few operations like credential providers and security providers for now and we can expand it later on?

Reply:
I plan on moving all the Provider classes (and the plugin itself) into a new module. You're welcome to do it if you want but this will be one of my next PRs once this is merged.